### PR TITLE
Model v4

### PR DESCRIPTION
**Pull request by bjrne**
Still needs testing and @mathebox's changes to the video entity.
Also paves the way for #208

### CODE REVIEW

**Comment by jprberlin** [comment]
We should also add completed flag for courses

File: xikolo-common/Data/Model/Enrollment+CoreDataProperties.swift [hunk 1]
```diff
@@ -21,6 +21,7 @@ extension Enrollment {
     @NSManaged var proctored_int: NSNumber?
     @NSManaged var completed_int: NSNumber?
```
Comment (bjrne) [inline xikolo-common/Data/Model/Enrollment+CoreDataProperties.swift]: Is this what you're looking for?

Comment (jprberlin) [inline xikolo-common/Data/Model/Enrollment+CoreDataProperties.swift]: hmm, jaa

**Comment by mathebox** [comment]
@bjrne Please have a look. This is kinda blocking.

File: xikolo-common/Data/xikolo.xcdatamodeld/xikolo 4.xcdatamodel/contents [hunk 2]
```diff
@@ -145,9 +145,11 @@
     <entity name="Video" representedClassName=".Video" parentEntity="Content" syncable="YES">
         <attribute name="audio_size" optional="YES" attributeType="Integer 32" defaultValueString="0" usesScalarValueType="YES" syncable="YES"/>
         <attribute name="audio_url" optional="YES" attributeType="Transformable" syncable="YES"/>
+        <attribute name="download_progress" optional="YES" attributeType="Decimal" syncable="YES"/>
```
Comment (bjrne) [inline xikolo-common/Data/xikolo.xcdatamodeld/xikolo 4.xcdatamodel/contents]: I don't think we ever used `Decimal` before, but I guess you tested it.

**Comment by mathebox** [comment]
Is there a reason why we use `String` to store the streams (video model) instead of `URL`?

**Comment by bjrne** [comment]
> Is there a reason why we use String to store the streams (video model) instead of URL?

There might be one, but I can't think of one right now.

**Comment by jprberlin** [comment]
Despite your comments lgtm. Feel free to merge when you want to and the discussed changes have been done

File: xikolo-common/Data/xikolo.xcdatamodeld/xikolo 4.xcdatamodel/contents (hunk 2 above)
Comment (mathebox) [inline xikolo-common/Data/xikolo.xcdatamodeld/xikolo 4.xcdatamodel/contents]: I just noticed `Decimal` has an unnecessary high precision..

**Pull request merged**